Initial interface

mecha opened the pull request:
Initial interfaces.

mecha committed:
Fixed autoload namespace

mecha committed:
Added composer scripts for convenience

mecha committed:
Added `dhii/stringable-interface` dep

mecha committed:
Added `BlockInterface`

mecha committed:
Added functional test for `BlockInterface`

XedinUnknown commented on src/BlockInterface.php:
```diff
@@ -0,0 +1,16 @@
+ *
+ * @since [*next-version*]
+ */
+interface BlockInterface extends StringableInterface
```
Shall we do the same thing as we discussed before, and give this interface a distinct method, such as `getOutput()`? Not sure if this is what we had agreed on, but if you agree with this - feel free to do that.

mecha replied:
@XedinUnknown I agree. What about `render()` instead?

XedinUnknown replied:
@mecha, `render()` sounds good.

mecha committed:
Added `BlockInterface#render()`

mecha committed:
Improved functional test

XedinUnknown commented on test/functional/BlockInterfaceTest.php:
```diff
@@ -59,11 +58,31 @@ public function testCanBeCreated()
+     *
+     * @since [*next-version*]
+     */
+    public function testRender()
```
Didn't really understand what you're testing here. This is an interface ;P

mecha replied:
Uhh ... haha. My bad. Brain derp 😆

mecha committed:
Removed test for `BlockInterface#render()`

mecha committed:
Added `dhii/output-renderer-interface` dependency

mecha committed:
Blocks are now renderers

XedinUnknown committed:
Updated dependencies

XedinUnknown committed:
Build fixing

Now building on Precise

XedinUnknown committed:
Documentation now more accurate and less precise for `BlockInterface`

XedinUnknown committed:
Standardized test

- Tests are now all unit, because interface; duh.
- FQNs fixed and standardized.
- Removed functionality test.
- Corrected error wording.

XedinUnknown committed:
Renamed to `output-block-interface`

mecha committed:
`BlockInterfaceTest` is now a unit test

mecha committed:
Added `BlockAwareInterface`

mecha committed:
Merge branch 'task/initial-interface' of https://github.com/Dhii/block-interface into task/initial-interface

# Conflicts:
#	test/unit/BlockInterfaceTest.php

mecha committed:
Fixed namespace to `Dhii\Output`

mecha committed:
Merge branch 'develop' into task/initial-interface

mecha committed:
Updated dependencies

mecha committed:
Merge branch 'task/initial-interface' of https://github.com/Dhii/output-block-interface into task/initial-interface

mecha commented:
@XedinUnknown The Packagist listing for this package needs to be updated after merging this PR. Any commit following the [rename commit](https://github.com/Dhii/output-block-interface/pull/1/commits/6446af143043ce5e1657d3a9e14835cb28b39b74) does not seem to be detected by Packagist - it only serves 0b6ab47 as the latest. This means that dependencies have to include the repository in their `composer.json` file to get the latest version.

mecha committed:
Blocks are no longer output renderers
